plugins/woocommerce/src/Internal/Admin/Logging/FileV2/File.php

```diff
@@ -0,0 +1,173 @@
+	}
+
+	/**
+	 * Parse the log filename to derive certain properties of the file.
+	 *
+	 * This makes assumptions about the structure of the log file's name. Using `-` to separate the name into segments,
+	 * if there are at least 5 segments, it assumes that the last segment is the "key" (i.e. hash), and the three
+	 * segments before that make up the date when the file was created in YYYY-MM-DD format. Any segments left after
+	 * that are the "source" that generated the log entries. If the filename doesn't have enough segments, it falls
+	 * back to the source and the key both being the entire filename, and using the last modified time as the creation
+	 * date.
+	 *
+	 * @return void
+	 */
+	protected function parse_filename(): void {
+		$info     = pathinfo( $this->path );
+		$filename = $info['filename'];
+		$segments = explode( '-', $filename );
+
+		if ( count( $segments ) >= 5 ) {
+			$this->source  = implode( '-', array_slice( $segments, 0, -4 ) );
+			$this->created = strtotime( implode( '-', array_slice( $segments, -4, 3 ) ) );
+			$this->hash    = array_slice( $segments, -1 )[0];
+		} else {
+			$this->source  = implode( '-', $segments );
+			$this->created = filemtime( $this->path );
+			$this->hash    = $this->source;
+		}
+
+		$rotation_marker = strrpos( $this->source, '.', -1 );
+		if ( false !== $rotation_marker ) {
+			$rotation = substr( $this->source, -1 );
+			if ( is_numeric( $rotation ) ) {
+				$this->rotation = intval( $rotation );
+			}
+
+			$this->source = substr( $this->source, 0, $rotation_marker );
+			if ( count( $segments ) < 5 ) {
+				$this->hash = $this->source;
+			}
+		}
+	}
+
+	/**
+	 * Get the name of the file, with extension, but without full path.
+	 *
+	 * @return string
+	 */
+	public function get_basename(): string {
+		return basename( $this->path );
+	}
+
+	/**
+	 * Get the file's source property.
+	 *
+	 * @return string
+	 */
+	public function get_source(): string {
+		return $this->source;
+	}
+
+	/**
+	 * Get the file's rotation property.
+	 *
+	 * @return int|null
+	 */
+	public function get_rotation(): ?int {
+		return $this->rotation;
+	}
+
+	/**
+	 * Get the file's hash property.
+	 *
+	 * @return string
+	 */
+	public function get_hash(): string {
+		return $this->hash;
+	}
+
+	/**
+	 * Get the file's created property.
+	 *
+	 * @return int|false
+	 */
+	public function get_created_timestamp() {
+		return $this->created;
+	}
+
+	/**
+	 * Get the time of the last modification of the file, as a Unix timestamp. Or false if the file isn't readable.
+	 *
+	 * @return int|false
+	 */
+	public function get_modified_timestamp() {
+		return filemtime( $this->path );
+	}
+
+	/**
+	 * Get the size of the file in bytes. Or false if the file isn't readable.
+	 *
+	 * @return int|false
+	 */
+	public function get_file_size() {
+		if ( ! is_readable( $this->path ) ) {
+			return false;
+		}
+
+		return @filesize( $this->path ); // phpcs:ignore WordPress.PHP.NoSilencedErrors.Discouraged
```

Comment:
WordPress core also does error suppression when calling functions like this. I'm assuming it's just to avoid fatal errors when dealing with the filesystem. I can imagine a scenario where the whole Logs tab is broken because there is somehow a corrupted file in the logs directory.

I am thinking about switching out the use of these functions directly and using `WP_Filesystem` instead, but I'm planning to look more into that in a later PR. The same goes for using `file_exists`. It might make sense to have the class call that when it first gets instantiated, but what should happen if the check fails? I'll look into that in a future PR as well.

Reply:
Yep, though, since use of `@` is a violation of WP coding standards I assume even they frown on its use, in general.

If the error is severe enough to result in a fatal error, my understanding is it does not stop the script from halting: it just means we won't know why it halted. Here talking about PHP < 8.0:

>  For example, prepending @ to a call of a function which did not exist, by being unavailable or mistyped, would cause the script to terminate with no indication as to why. 

I guess that *could* be argued to be a good thing in this scenario. Anyway, your plan to revisit later sounds good to me :+1: